vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/PoolImpl.java

```diff
@@ -155,6 +157,24 @@ public Future<SqlConnection> getConnection() {
     });
   }
 
+  @Override
+  public Future<SqlConnection> getPropagatableConnection() {
+    if (propagatableConnection == null) {
```

Comment:
the idea is that such transaction is stored in the Vert.x current context and not a field on the pool, allowing multiple concurrent transactions.

Reply:
I somehow managed to forget that completely. 
There weren't really that many examples showing how items should be managed in the context, so please let me know if this is not the way you prefer doing this.

I've also cleaned up the code a bit, avoiding unneccesary additional methods here and there.

Reply:
it should use Vert.x `Context` with getLocal/putLocal methods

Reply:
Changed it now. Is it safe to assume there always being a current context available?

Reply:
yes, it is lazy created when it does not exist